src/replicator.rs

```diff
@@ -195,10 +218,25 @@ impl Replicator {
         );
 
         info!("window created, waiting for ledger download done");
+        let start = Instant::now();
+        let mut received_so_far = 0;
+
         while !done.load(Ordering::Relaxed) {
             sleep(Duration::from_millis(100));
+
+            let elapsed = start.elapsed();
+            received_so_far += entry_rx.try_recv().map(|v| v.len()).unwrap_or(0);
```

Comment:
Seems like we will drop the receiver after this new(), will we leak entries into in the channel or will an error on the `send` occur?

Reply:
send will start producing errors once the receiver is dropped. I know that these `mpsc` channels have an unbounded buffer, but things only get pushed onto it if both ends are connected. I'm not sure how the sender on the other side will be affected by producing errors, I will look into that next. 

If this is a problem, is adding the receiver as a member of the `Replicator` struct a reasonable solution.

Reply:
Yea, adding it to the Replicator struct should be fine.